packages/wallets/src/components/WalletModal/WalletModal.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { ComponentPropsWithoutRef, PropsWithChildren, useRef } from 'react';
+import { useOnClickOutside } from 'usehooks-ts';
+import CloseIcon from '../../public/images/ic-close-dark.svg';
+import { useModal } from '../ModalProvider';
+
+type TWalletModalProps = {
+    should_show_close_icon?: boolean;
+    should_close_on_click_outside?: boolean;
+    onClickCloseIcon?: () => void;
+};
+
+const WalletModal = ({
+    children,
+    onClickCloseIcon,
+    should_show_close_icon,
+    should_close_on_click_outside = true,
+    className: propsClassName,
+    ...rest
+}: ComponentPropsWithoutRef<'div'> & PropsWithChildren<TWalletModalProps>) => {
+    const { hide } = useModal();
+    const modalRef = useRef(null);
+
+    useOnClickOutside(modalRef, should_close_on_click_outside ? hide : () => undefined);
```

Comment:
eslint was complaining that i cannot put empty function `() => {}`, so I put `() => undefined` instead 😢

Reply:
I think it is becasue of esLint rule
The silent error handling that it should explicitly return something instead of just inferring at runtime.
cc:@farzin-deriv